src/ts/editorcanvas/CanvasVM.ts

```diff
@@ -38,6 +38,8 @@ module editorcanvas {
         private scrollDiv : HTMLDivElement;
         private properties : CanvasProperties = new CanvasProperties();
         private eventsGivenToCanvas : string = "click mousedown mouseup mousemove";
+        private numAssetsToLoad : number = 0;
```

Comment:
It feels weird to have the asset logic embedded directly into the CanvasVM class.